controlplane/kubeadm/controllers/scale.go

```diff
@@ -239,5 +239,8 @@ func selectMachineForScaleDown(controlPlane *internal.ControlPlane, outdatedMach
 	if outdatedMachines.Len() > 0 {
 		machines = outdatedMachines
 	}
+	if withDeleteAnnotation := controlPlane.MachineWithDeleteAnnotation(machines); withDeleteAnnotation.Len() > 0 {
+		machines = withDeleteAnnotation
+	}
```

Comment:
Is the intention here to prioritize outdated Machines over Machines marked for deletion by annotation?

This would mean effectively that the annotation would be ignored if applied to a newer machine during the process of a rollout of changes until that rollout is complete.

I'm wondering if this will cause confusion for users.

I think it might make more sense to prioritize:
- any outdatedMachines with the delete annotation
- any machines with the delete annotation
- outdated machines
- all machines

```suggestion
	switch {
	case controlPlane.MachineWithDeleteAnnotation(outdatedMachines).Len() > 0:
		machines = controlPlane.MachineWithDeleteAnnotation(outdatedMachines)
	case controlPlane.MachineWithDeleteAnnotation(machines).Len() > 0:
		machines = controlPlane.MachineWithDeleteAnnotation(machines)
	case outdatedMachines.Len() > 0:
		machines = outdatedMachines
	}
```

Reply:
That should be what the current code does, given that we replace the splice in each `if` block, but your switch case solution looks more readable to me 👍

Reply:
@vincepri there is a slightly different effect if the delete annotation is added to a machine that is up to date according to our rollout policies while there are any outdated machines that still exist.

The prioritization of the current code in the PR is:
- any outdatedMachines with the delete annotation
- outdated machines
- any machines with the delete annotation
- all machines

compared to my suggestion:
- any outdatedMachines with the delete annotation
- any machines with the delete annotation
- outdated machines
- all machines

Reply:
Oh I see what you mean, yes that makes sense! ✅

Reply:
That makes sense to me as well, thanks for suggestion @detiber. I have addressed your comments. The original idea was giving priority to any machines(outdatedMachines or not) with annotation a top priority, as you have clarified above.  WDYT of this version? PTAL.